features/measurements/impl/src/test/java/org/opennms/netmgt/measurements/impl/JEXLExpressionEngineEnhancedTest.java

```diff
@@ -0,0 +1,307 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2010-2015 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2015 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.measurements.impl;
+
+import static org.junit.Assert.*;
+
+import java.util.Map;
+
+import org.apache.commons.jexl2.JexlContext;
+import org.junit.Test;
+import org.opennms.netmgt.measurements.api.ExpressionEngine;
+import org.opennms.netmgt.measurements.api.exceptions.ExpressionException;
+import org.opennms.netmgt.measurements.api.FetchResults;
+import org.opennms.netmgt.measurements.model.Expression;
+import org.opennms.netmgt.measurements.model.QueryRequest;
+import org.opennms.netmgt.measurements.model.Source;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+public class JEXLExpressionEngineEnhancedTest {
+
+	private final ExpressionEngine jexlExpressionEngine = new JEXLExpressionEngine();
+
+	@Test
+	public void checkSamplefn5NaN(){
+		final String expression = "fn:arrayNaN(\"x\", 5)";
+		boolean success = true;
+
+		StringBuffer sb = new StringBuffer("");
+		try{
+
+			double[] result = performExpression(expression);
+
+			for (double value: result){
+				sb.append(value+",");
+			}
+
+			success = 
+					Double.isNaN(result[0]) &&
```

Comment:
If the test were to fail, the output gives no indication as to what the expected values are. For this reason I prefer using asserts on the values were checking i.e. `assertEquals(Double.NaN, result[0], delta);` instead of tracking them via boolean variables.